persistence/sql/persister_identity.go

```diff
@@ -460,17 +462,81 @@ func (p *Persister) GetIdentity(ctx context.Context, id uuid.UUID, expand identi
 		attribute.String("network.id", p.NetworkID(ctx).String()),
 	)
 
+	nid := p.NetworkID(ctx)
 	con := p.GetConnection(ctx)
-	query := con.Where("id = ? AND nid = ?", id, p.NetworkID(ctx))
-	if len(expand) > 0 {
-		query = query.EagerPreload(expand.ToEager()...)
-	}
+	query := con.Where("id = ? AND nid = ?", id, nid)
+
+	var (
+		i                   identity.Identity
+		credentials         []identity.Credentials
+		verifiableAddresses []identity.VerifiableAddress
+		recoveryAddresses   []identity.RecoveryAddress
+	)
 
-	var i identity.Identity
 	if err := query.First(&i); err != nil {
 		return nil, sqlcon.HandleError(err)
 	}
 
+	eg, ctx := errgroup.WithContext(ctx)
+	if expand.Has(identity.ExpandFieldRecoveryAddresses) {
+		eg.Go(func() error {
+			// We use WithContext to get a copy of the connection struct, which solves the race detector
+			// from complaining incorrectly.
+			//
+			// https://github.com/gobuffalo/pop/issues/723
+			if err := con.WithContext(ctx).
+				Where("identity_id = ? AND nid = ?", i.ID, nid).
+				Order("id ASC").
+				All(&recoveryAddresses); err != nil {
+				return sqlcon.HandleError(err)
+			}
+			return nil
+		})
+	}
+
+	if expand.Has(identity.ExpandFieldVerifiableAddresses) {
+		eg.Go(func() error {
+			// We use WithContext to get a copy of the connection struct, which solves the race detector
+			// from complaining incorrectly.
+			//
+			// https://github.com/gobuffalo/pop/issues/723
+			if err := con.WithContext(ctx).
+				Order("id ASC").
+				Where("identity_id = ? AND nid = ?", i.ID, nid).All(&verifiableAddresses); err != nil {
+				return sqlcon.HandleError(err)
+			}
+			return nil
+		})
+	}
+
+	if expand.Has(identity.ExpandFieldCredentials) {
+		eg.Go(func() error {
+			// We use WithContext to get a copy of the connection struct, which solves the race detector
+			// from complaining incorrectly.
+			//
+			// https://github.com/gobuffalo/pop/issues/723
+			if err := con.WithContext(ctx).
+				EagerPreload("IdentityCredentialType", "CredentialIdentifiers").
```

Comment:
This could further be optimized. `IdentityCredentialType` is basically a static value that never changes, even across different nids (iirc). It could thus be cached, saving us a whole query (and busy connection).